test/test_auth.py

```diff
@@ -340,6 +340,70 @@ def test_auth_dbname_usage_global_setting(
             pass
 
 
+@pytest.mark.skipif("WINDOWS", reason="Windows does not have SIGHUP")
+def test_auth_query_database_setting(
+    bouncer,
+):
+    """
+    Check the pgbouncer can use auth_query in database section to get password
+    """
+
+    config = f"""
+        [databases]
+        postgres = auth_query='SELECT usename, passwd FROM pg_shadow where usename = $1'\
+            host={bouncer.pg.host} port={bouncer.pg.port}
+        [pgbouncer]
+        auth_query = SELECT usename, passwd FROM pg_shadow where usename = $1
```

Comment:
This is the same as the one in the databases section. Let's change it to something useless like `SELECT 1`.

Reply:
Don't we want to test that it works? If so, I suggest that a functional test is added. Create a table, insert one row and test if the authentication works.

```
create table auth_query_table (u text, p text);
insert into auth_query_table VALUES('u0', 'p0');
```

I would also suggest that the used auth_query is exposed for debug purposes.

```
-   slog_noise(client, "doing auth_conn query");
+   slog_noise(client, "doing auth_conn query: %s", auth_query);
```

Reply:
> Don't we want to test that it works?

Your proposed test would work too afaict. But with the current test we test that setting auth_query in the database section has effect. i.e. if it didn't have any effect, then the global broken one would be used and you'd get an error like this in the logs:
```
BOUNCER_LOG
2023-11-03 16:45:42.916 CET [1845812] LOG S-0x5633b3d02dd0: postgres/pswcheck@127.0.0.1:10202 closing because: error response from auth_query (age=0s)
2023-11-03 16:45:42.916 CET [1845812] LOG C-0x5633b3cf5f50: postgres/(nouser)@127.0.0.1:53118 closing because: bouncer config error (age=0s)

PG_LOG
2023-11-03 16:45:42.916 CET [1845814] ERROR:  bind message supplies 1 parameters, but prepared statement "" requires 0
2023-11-03 16:45:42.916 CET [1845814] STATEMENT:  SELECT 1
``` 

So I think the current test is is acceptable at least.

Reply:
WFM.